### PR TITLE
Added intro and background for Reactor Pattern Guide

### PR DESCRIPTION
**Pull request by anfeldma-ms**

### CODE REVIEW

File: reactor-pattern-guide.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Guide to Reactive Programming Programming and Reactor Design Patterns with Azure Cosmos DB Java SDK
+
+## Introduction
+
+The purpose of this document is to enable
+* users with RxJava experience (i.e. from working with Cosmos DB Async Java SDK v2.x.x)
```
Comment (moderakh) [inline reactor-pattern-guide.md]: I think there should be a "," after i.e. -> i.e., 
no?

File: reactor-pattern-guide.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+# Guide to Reactive Programming Programming and Reactor Design Patterns with Azure Cosmos DB Java SDK
```
Comment (moderakh) [inline reactor-pattern-guide.md]: as all azure SDKs are using reactive stream. it is worth mentioning that all azure sdks will be using reactive stream. Also probably there should be some general doc from azure-sdk-for-java on reactive stream usage, if so you may want to link it here.

File: reactor-pattern-guide.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+| 3.x.x                  | Project Reactive |
+| 4.x.x                  | Project Reactive | 
+
+
+
+## Background: Reactive Programming, Reactive Streams, Reactor, Rx Java, and Project Reactive
+
+### 1. ***Reactive Programming and Standards***
+
+Reactive Programming is a declarative programming paradigm in which program operation and control flow are described as a stream of data items passing through a pipeline of operations in which each operation affects the data which flows downstream.
```
Comment (moderakh) [inline reactor-pattern-guide.md]: if there is a high level guide on azure sdk use of reactive stream (all track 2 azure sdks will be using reactive streams) it should be linked here.

File: reactor-pattern-guide.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+# => Functionality
+
+# => Terminology
+
+# Diagrams
+
+# Tips and Tricks
+
+# Troubleshooting
```
Comment (moderakh) [inline reactor-pattern-guide.md]: I am not sure, if troubleshooting guide should be added to samples.

We have a separate troubleshooting guide. maybe just a link reference suffices.

File: reactor-pattern-guide.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+# Hazards
+
+C10K problem
```
Comment (moderakh) [inline reactor-pattern-guide.md]: what's this?

Comment (anfeldma-ms) [reply reactor-pattern-guide.md]: https://en.wikipedia.org/wiki/C10k_problem

**Pull request merged**